backend/inspirehep/cds/tasks.py

```diff
@@ -0,0 +1,152 @@
+from inspirehep.cds.models import CDSRun, CDSRunStatus
+from inspirehep.records.api import LiteratureRecord
+
+LOGGER = structlog.getLogger()
+
+MAX_RETRY_COUNT = 3
+RETRY_BACKOFF = 5
+CHUNK_SIZE = 100
+
+
+@shared_task(
+    bind=True,
+    ignore_results=False,
+    queue="migrator",
+    acks_late=True,
+    retry_backoff=RETRY_BACKOFF,
+    retry_kwargs={"max_retries": MAX_RETRY_COUNT},
+    autoretry_for=(ConnectionError, RequestException),
+)
+def sync_identifiers(self, since=None):
+    if not current_app.config.get("FEATURE_FLAG_ENABLE_CDS_SYNC", False):
+        LOGGER.warning("CDS sync is currently disabled by feature flag.")
+        return
+
+    last_run = CDSRun.get_last_successfull_run()
+    last_run_date = last_run.date if last_run else None
+
+    since = since or last_run_date
+    if not since:
+        LOGGER.error(
+            "CDS Sync failed. No `since` provided and no successful runs in DB. Aborting."
+        )
+        raise CDSSyncError("Missing `since` date")
+
+    CDSRun.new_run(self)
+    try:
+        run_sync(since)
+    except Exception as err:
+        LOGGER.exception("CDS Sync failed", error=err)
+        CDSRun.update_status(self, status=CDSRunStatus.ERROR, message=str(err))
+        raise
+    else:
+        CDSRun.update_status(self, status=CDSRunStatus.FINISHED)
+
+
+def run_sync(since):
+    LOGGER.info("Starting CDS identifiers sync", since=since)
+    cds_api = current_app.config.get("CDS_SERVER_API")
+    service = "inspire2cdsids"
+    if isinstance(since, datetime):
+        since = since.date()
+    elif not isinstance(since, date):
+        try:
+            since = date.fromisoformat(since)
+
+        except ValueError:
+            raise WrongDateFormat(
+                f"`since`: {since} is in wrong format. Should be in ISO format: YYYY-MM-DD."
+            )
+
+    if not cds_api:
+        raise MissingCDSServerConfig("CDS Server config is missing.")
+    url = parse.urljoin(cds_api, service)
+    response = requests.get(url, params={"since": since.isoformat()})
+    response.raise_for_status()
+    process_cds_response(response.json())
+
+
+def process_cds_response(response):
+    try:
+        for idx, record in enumerate(response.get("hits", [])):
+            process_cds_record(record)
+            if idx % CHUNK_SIZE:
+                db.session.commit()
+        db.session.commit()
+
+    except SQLAlchemyError as err:
+        db.session.rollback()
+        LOGGER.exception("Cannot commit a batch of updates from CDS.", error=err)
+        raise
+    except Exception:
+        db.session.rollback()
+        raise
+
+
+def process_cds_record(cds_record):
+    control_number = get_value(cds_record, "metadata.other_ids[0]")
+    cds_id = cds_record.get("id") or get_value(cds_record, "metadata.control_number")
+
+    if not control_number:
+        LOGGER.error(
+            "Cannot extract control_number from CDS response", cds_data=cds_record
```

Comment:
I think this is a misunderstanding. We don't want to match only by control number. We also want to match by arXiv ID, DOI, or report number (if unique).